hpx/runtime/threads/detail/thread_num_tss.hpp

```diff
@@ -10,20 +10,30 @@
 #include <hpx/config.hpp>
 
 #include <cstddef>
+#include <cstdint>
+#include <tuple>
+#include <utility>
 
 #include <hpx/config/warnings_prefix.hpp>
 
-namespace hpx { namespace threads { namespace detail
-{
+namespace hpx { namespace threads { namespace detail {
+    // set/get the global thread Id to/from thread local storage
     HPX_EXPORT std::size_t set_thread_num_tss(std::size_t num);
     HPX_EXPORT std::size_t get_thread_num_tss();
 
+    // this tuple holds the local thread Id and the pool index
+    // associated with the thread
+    using thread_tuple = std::tuple<std::uint16_t, std::uint16_t>;
```

Comment:
Any reason not to use `hpx::util::tuple`? Also, using tuple's in contexts like this usually suggests some kind of 'laziness' on behalf of the programmer. An explicit struct would probably be more descriptive.

Reply:
I didn;t realize tuples were the lazy man's struct! I will use a struct and give the members proper names to make things explicit. If it doesn't add overheads when fetching the data (I think it is returned by ref), I'll combine all 3 into one struct and unify them.